cli/joy.js

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env node
+'use strict';
+const fs = require('fs');
+const prog = require('caporal');
+const path = require('path');
+const pa11y = require('pa11y');
+const Docker = require('./plugins/docker');
+const Test = require('./plugins/test');
+const util = require('util');
+
+class Joy {
+  constructor() {
+    this.env = process.env;
+    this.prog = prog;
+    this.prog.version('1.0.0');
+  }
+
+  addCommand(commandObj) {
+    new commandObj(this);
+  }
+
+
+  /**
+   * Indicates whether the CWD is a Joy enabled project (e.g. is the parent to a .joy subdirectory)
+   * 
+   */
+  isJoy() {
+    try {
+      const stats = fs.statSync(path.join(process.cwd(), './.joy'));
+      return stats && stats.isDirectory();
+    }
+    catch (e) {
+      return false;
+    }
+  }
+
+
+  /**
+   * Simple wrapper to exec a child process to call an external binary. 
+   * Child stdout and stderror are piped directly to ours.
+   * Returns a shell success code
+   * 
+   * @param {*} cmd 
+   * @param {*} params 
+   */
+  invoke(cmd, params) {
+    return new Promise((resolve, reject) => {
+      const child = require('child_process').spawn(cmd, params);
+
+      // Connect the child's std* to our std*
+      child.stdout.pipe(process.stdout);
+      child.stderr.pipe(process.stderr);
+
+      child.on('close', code => {
```

Comment:
I usually try to wrap the single parameters in parentheses for consistency and for cleaner code commits when adding additional parameters (edge case).

Reply:
Good call. All of these so far are eslint'able. I thought I had them all enabled. Will check.